common/src/main/java/net/adoptopenjdk/icedteaweb/http/CloseableHttpConnection.java

```diff
@@ -0,0 +1,37 @@
+package net.adoptopenjdk.icedteaweb.http;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.net.HttpURLConnection;
+
+/**
+ * {@link Closeable} wrapper around a {@link HttpURLConnection}.
+ *
+ * Only the needed methods of HttpURLConnection are exposed.
+ *
+ * Closing the connection will trigger a disconnect on the underlying HttpUrlConnection
+ */
+public class CloseableHttpConnection extends CloseableConnection {
+    private final HttpURLConnection delegate;
```

Comment:
maybe we get define a protected getter in `CloseableConnection` instead of having it as a field twice

Reply:
if I would add a getter or field I would have 2 side effects:
- need for casting or introducing a generic parameter
- opening access to to the wrapped connection at least within the package

Reply:
ah, have not seen that it's not the same type ;)

Reply:
I would go with generics than but I'm fine with your solution, too.